src/ID3Util.ts

```diff
@@ -11,34 +11,25 @@ export class SplitBuffer {
     }
 }
 
+/**
+ * Expects a buffer containing a string at the beginning that is terminated by a \0 character.
+ * Returns a split buffer containing the bytes before and after null termination.
+ */
 export function splitNullTerminatedBuffer(buffer: Buffer, encodingByte = 0x00) {
-    const termination = { start: -1, size: 0 }
-    if (encodingByte === 0x01 || encodingByte === 0x02) {
-        termination.start = buffer.indexOf(Buffer.from([0x00, 0x00]))
-        termination.size = 2
-        if (
-            termination.start !== -1 &&
-            buffer.length > (termination.start + termination.size)
-        ) {
-            if (buffer[termination.start + termination.size] === 0x00) {
-                termination.start += 1
-            }
+    // UTF-16/BE always uses two bytes per character.
+    // \0 is therefore encoded as [0x00, 0x00] instead of just [0x00].
+    // We'll do a sliding window search, window size depends on encoding.
+    const charSize = [0x01, 0x02].includes(encodingByte) ? 2 : 1
+    for(let pos = 0; pos + charSize - 1 < buffer.length; pos += charSize) {
```

Comment:
I'm not sure about `+charSize-1` I need to go, will get back to it soon.

Reply:
Adding charSize is needed because otherwise it would overflow if a broken buffer is passed e.g. [0xff, 0xfe, 0x00]
but this would cause problems if charSize is 1 because then the 0x00 would be skipped (2 + 1 < 3), that's why the for loop should only take the additional characters into account (charSize > 1).

Reply:
OK, I see good catch, I did not think about this case.
This way of handling it is counter intuitive.
Looking at the documentation of `Buffer.readUIntBE` this is expressed differently.

```
@param offset — Number of bytes to skip before starting to read. Must satisfy 0 <= offset <= buf.length - byteLength.

@param byteLength — Number of bytes to read. Must satisfy 0 < byteLength <= 6.
```

So I suggest to use the same algorithm i.e.
```ts
for (let pos = 0; pos <= buffer.length - charSize; pos += charSize) {
```